### PR TITLE
feat: refactor best/worst record api in response time

### PR DESCRIPTION
**Pull request by rokamu623**
## Description
The method of calculating ResponseTime for the `best` and `worst-with-external-latency` cases has been changed to be the same as for `all` and `worst` case.

## Related links
Jira: https://tier4.atlassian.net/browse/RT2-889

## Pre-review checklist for the PR author

- [x] I've confirmed the [contribution guidelines](https://github.com/tier4/caret/blob/main/.github/CONTRIBUTING.md).
- [x] The PR is ready for review.

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR has been properly tested.
- [x] The PR has been reviewed.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] (Optional) The PR has been properly tested with CARET_report verification.
- [x] There are no open discussions or they are tracked via tickets.
- [x] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

### CODE REVIEW

File: src/test/record/records_service/test_response_time_all.py [hunk 1]
```diff
@@ -139,6 +139,383 @@ def test_drop_case(self):
...
+        records = create_records(records_raw, columns)
+        response = ResponseTime(records)
+
+        response_time = response.to_best_case_records()
+        expect = [
+            {'start': 3, 'response_time': 1},
+            {'start': 4, 'response_time': 6},
+            {'start': 6, 'response_time': 0}
+        ]
+        assert to_dict(response_time) == expect
```
Comment (bopeng-saitama) [inline src/test/record/records_service/test_response_time_all.py:269]: @rokamu623 
Is it poasible that the start_ts is bigger than end_ts?
If the case is possible, I think you should add the test.

**Review by bopeng-saitama — APPROVED** [review]
LGTM

**Review by xygyo77 — COMMENTED** [review]
I used "autoware_launch_trace_20221227-150253_universe_rosbag" to get the timeseries graph of ResponseTime before and after applying this PR. Before applying this PR, the version was bokeh3.3.0.
Visual inspection showed that there were differences in best and worst-with-external-latency. Please check the difference.
The first four graphs are before application of PR and the next four are after.

PR323-pre-resp_tl_best_bokeh_plot
PR323-pre-resp_tl_worst_bokeh_plot
PR323-pre-resp_tl_all_bokeh_plot
PR323-pre-resp_tl_wwel_bokeh_plot
PR323-after-resp-tl-best-bokeh_plot
PR323-after-resp-tl-worst-bokeh_plot
PR323-after-resp-tl-all-bokeh_plot
PR323-after-resp-tl-wwel-bokeh_plot

**Comment by rokamu623** [comment]
@xygyo77 
Is the difference about `best-case`?

Regarding this,
Before refactoring, the response time of the 1st path is not calculated.
This is because in the `worst-with-external-latency case`, the start time of the previous path (worst-case external input time) cannot be defined.

In fact, except in the `worst-with-external-latency case`, the response time can be calculated for the 1st path (output time - input time).

Therefore, after refactoring, the response time for the 1st path is recorded more than before refactoring.

**Review by xygyo77 — APPROVED** [review]
LGTM
Note: With [https://tier4.atlassian.net/browse/RT2-888](https://tier4.atlassian.net/browse/RT2-888), the "worst-with-external-latency" case has been replaced by the old "worst" case.

**Pull request merged**